homeassistant/components/climate/touchline.py

```diff
@@ -0,0 +1,89 @@
+"""
+Platform for Roth Touchline heat pump controller.
+
+Example configuration:
+climate:
+  - platform: touchline
+    host: http://192.168.1.1
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/climate.touchline/
+"""
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.climate import ClimateDevice, PLATFORM_SCHEMA
+from homeassistant.const import CONF_HOST, TEMP_CELSIUS, ATTR_TEMPERATURE
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['pytouchline==0.6']
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Touchline devices."""
+    from pytouchline import PyTouchline
+    host = config[CONF_HOST]
+    py_touchline = PyTouchline()
+    number_of_devices = int(py_touchline.get_number_of_devices(host))
+    devices = []
+    for device_id in range(0, number_of_devices):
+        devices.append(Touchline(PyTouchline(device_id)))
+    add_devices(devices)
+
+
+class Touchline(ClimateDevice):
+    """Representation of a Touchline device."""
+
+    def __init__(self, touchline_thermostat):
+        """Initialize the climate device."""
+        self.unit = touchline_thermostat
+        self.unit.update()
```

Comment:
Don't do this here. Call `add_devices` above with `update=True`. This causes HA to schedule a call to your `update()` method as soon as the device was added.